### PR TITLE
Allow setting eip 1559 fee params

### PR DESCRIPTION
**Pull request by mycodecrafting**
Adds a per block height config to set some EIP1559 fee params (elasticity and denominator) + adds a min base fee parameter.

### CODE REVIEW

**Review by joroshiba — APPROVED** [review]
LGTM with couple nits.

File: genesis.json [hunk 1]
```diff
@@ -21,7 +21,10 @@
         "astriaCelestiaInitialHeight": 2,
         "astriaCelestiaHeightVariance": 10,
         "astriaBridgeAddresses": [],
-        "astriaBridgeAllowedAssetDenom": "nria"
+        "astriaBridgeAllowedAssetDenom": "nria",
+        "astriaEIP1559Params": {
+            "1": { "minBaseFee": 0, "elasticityMultiplier": 2, "BaseFeeChangeDenominator": 8 }
```
[the remark this answers is not in the document]
Comment (mycodecrafting) [reply genesis.json:26]: ya it has to be a string cause json key names can only be strings.

File: params/config.go [hunk 2]
```diff
@@ -368,6 +371,72 @@ func (c *ChainConfig) AstriaExtraData() []byte {
...
+
+func (c AstriaEIP1559Params) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.heights)
+}
+
+func (c *AstriaEIP1559Params) UnmarshalJSON(data []byte) error {
+	if err := json.Unmarshal(data, &c.heights); err != nil {
+		return err
+	}
+	c.init()
```
[the remark this answers is not in the document]
Comment (mycodecrafting) [reply params/config.go:436]: New function makes more sense and I think better covers the case of manual instantiation (hard coded genesis params).

[the remark this answers is not in the document]
Comment (mycodecrafting) [reply params/config.go]: updated

**Review by noot — APPROVED** [review]

File: params/config.go [hunk 3]
```diff
@@ -368,6 +371,70 @@ func (c *ChainConfig) AstriaExtraData() []byte {
...
+func (c AstriaEIP1559Params) MarshalJSON() ([]byte, error) {
+	return json.Marshal(c.heights)
+}
+
+func (c *AstriaEIP1559Params) UnmarshalJSON(data []byte) error {
+	var heights map[uint64]AstriaEIP1559Param
+	if err := json.Unmarshal(data, &heights); err != nil {
+		return err
+	}
+	*c = *NewAstriaEIP1559Params(heights)
```
Comment (noot) [inline params/config.go:434]: does this actually work? or do you have to set each field within `c` manually? can you test this

Comment (mycodecrafting) [reply params/config.go:434]: ya it works. I made a test script and it worked. I can add a few tests tho, prob not a bad idea.

**Pull request merged**